plugin/src/main/java/hudson/plugins/swarm/PluginImpl.java

```diff
@@ -54,9 +57,28 @@ public void doCreateSlave(StaplerRequest req, StaplerResponse rsp, @QueryParamet
 				nodeProperties = Lists.newArrayList(new ToolLocationNodeProperty(parsedToolLocations));
             }
     		
-            // try to make the name unique. Swarm clients are often replicated VMs, and they may have the same name.
-            if (jenkins.getNode(name) != null) {
-                name = name + '-' + req.getRemoteAddr();
+            if (hash == null && jenkins.getNode(name) != null) {
+                // this is a legacy client, they won't be able to pick up the new name, so throw them away
+                // perhaps they can find another master to connect to
+                rsp.setStatus(SC_CONFLICT);
```

Comment:
I think it would be wise to include a plain text response here for the poor admin that tries to debug this.